typescript/cli/src/tests/commands/core.ts

```diff
@@ -1,117 +1,163 @@
 import { $, ProcessPromise } from 'zx';
 
 import { DerivedCoreConfig } from '@hyperlane-xyz/sdk';
-import { Address } from '@hyperlane-xyz/utils';
+import { ProtocolType } from '@hyperlane-xyz/utils';
 
 import { readYamlOrJson } from '../../utils/files.js';
 
-import { ANVIL_KEY, REGISTRY_PATH, localTestRunCmdPrefix } from './helpers.js';
-
-/**
- * Deploys the Hyperlane core contracts to the specified chain using the provided config.
- */
-export function hyperlaneCoreDeployRaw(
-  coreInputPath: string,
-  privateKey?: string,
-  skipConfirmationPrompts?: boolean,
-  hypKey?: string,
-): ProcessPromise {
-  return $`${
-    hypKey ? ['HYP_KEY=' + hypKey] : []
-  } ${localTestRunCmdPrefix()} hyperlane core deploy \
-        --registry ${REGISTRY_PATH} \
-        --config ${coreInputPath} \
-        ${privateKey ? ['--key', privateKey] : []} \
-        --verbosity debug \
-        ${skipConfirmationPrompts ? ['--yes'] : []}`;
-}
+import { localTestRunCmdPrefix } from './helpers.js';
 
-/**
- * Deploys the Hyperlane core contracts to the specified chain using the provided config.
- */
-export async function hyperlaneCoreDeploy(
-  chain: string,
-  coreInputPath: string,
-) {
-  return $`${localTestRunCmdPrefix()} hyperlane core deploy \
-        --registry ${REGISTRY_PATH} \
-        --config ${coreInputPath} \
-        --chain ${chain} \
-        --key ${ANVIL_KEY} \
-        --verbosity debug \
-        --yes`;
-}
+export class HyperlaneCore {
```

Comment:
I would change this name to `HyperlaneE2ECoreTestCommads` as it might be easy to think that this class allows to handle core deployments